haralyzer/http.py

```diff
@@ -262,25 +266,34 @@ def lastModified(self) -> str:
         return self.get_header_value("last-modified")
 
     @cached_property
-    def mimeType(self) -> str:
+    def mimeType(self) -> Optional[str]:
         """
         :return: Mime Type of response
         :rtype: str
         """
-        return self.raw_entry["content"]["mimeType"]
+        if "content" not in self.raw_entry:
+            return None
+        return self.raw_entry["content"].get("mimeType")
 
     @cached_property
-    def text(self) -> str:
+    def text(self) -> Optional[str]:
         """
         :return: Response body
         :rtype: str
         """
-        return self.raw_entry["content"]["text"]
+        if "content" not in self.raw_entry:
+            return None
+        content = self.raw_entry["content"]
+        if "_textBase64" in content:

```

Comment:
What is this all about? Why would we prefer base 64 encoding over just the `text` attribute?

Reply:
Hey,
The above fields cannot co-exist. Therefore if there is `_textBase64 ` the field `text` won't be exists and vice versa.

I have noticed that `harlayzer` failed to parse `postData` correctly since `_textBase64`  was not taking into account.

Not sure if it's part of HAR specifications but this format used a lot by Apple.

```json
"request": {
    "method": ,
    "bodySize": ,
    "headersSize": ,
    "postData": {
        "_textBase64":"" ,
        "mimeType": ""
    }
```

Reply:
It doesn't look like `_textBase64` is a part of the [spec](https://w3c.github.io/web-performance/specs/HAR/Overview.html).
> This object describes posted data, if any (embedded in <request> object).
```
This object describes posted data, if any (embedded in <request> object).
"postData": {
    "mimeType": "multipart/form-data",
    "params": [],
    "text" : "plain posted data",
    "comment": ""
}

    mimeType [string] - Mime type of posted data.
    params [array] - List of posted parameters (in case of URL encoded parameters).
    text [string] - Plain text posted data
    comment [string, optional] (new in 1.2) - A comment provided by the user or the application.
```
If this is something that is specific to Apple's HAR format, then I would propose a backwards compatible way of handling that.

Reply:
Removing the `KeyError` isn't backwards compatible, as it will break code which relies on it being thrown. Is it specifically the Safari browser that is generating a HAR format that doesn't use the `content` key?

Reply:
About the HAR format, It's system wide generated by `CFNetwork`.

I restored the original behavior of the `KeyError`.

Reply:
Interesting. Is there a spec for how `CFNetwork` formats HAR files?

@mrname I think this looks like a better way to handle the encoding

Reply:
Unfortunately, I don't have the specs as Apple did not publish them. The entries I am trying to parse are sent as part of the `com.apple.os_trace_relay` log messages. You can review them using one of our tools [pymobiledevice3](https://github.com/doronz88/pymobiledevice3) | [harlogger](https://github.com/doronz88/harlogger) or via Apple's [console.app](https://support.apple.com/en-il/guide/console/welcome/mac)